pygrocy/grocy_api_client.py

```diff
@@ -327,39 +327,61 @@ def __init__(self, base_url, api_key, port: int = DEFAULT_PORT_NUMBER, verify_ss
                 "GROCY-API-KEY": api_key
             }
 
+    def _do_request(self, request_type: str, end_url: str, data = None):
+        req_url = urljoin(self._base_url, end_url)
+        if request_type == "GET":
+            resp = requests.get(
+                req_url, verify=self._verify_ssl, headers=self._headers)
+        if request_type == "POST":
+            if data:
+                resp = requests.post(
+                    req_url, verify=self._verify_ssl,
+                    headers=self._headers,
+                    data=data)
+        if request_type == "PUT":
+            if data:
+                up_header = self._headers.copy()
+                up_header['accept'] = '*/*'
+                if isinstance(data, dict):
+                    up_header['Content-Type'] = 'application/json'
+                    data = json.dumps(data)
+                else:
+                    up_header['Content-Type'] = 'application/octet-stream'
+                resp = requests.put(
+                    req_url, verify=self._verify_ssl,
+                    headers=up_header,
+                    data=data)
+        resp.raise_for_status()
+        if len(resp.content) > 0:
+            return resp.json()
+
+
     def get_stock(self) -> List[CurrentStockResponse]:
-        req_url = urljoin(self._base_url, "stock")
-        resp = requests.get(req_url, verify=self._verify_ssl, headers=self._headers)
-        if resp.status_code != 200 or not resp.text:
-            return
-        parsed_json = resp.json()
-        return [CurrentStockResponse(response) for response in parsed_json]
+        parsed_json = self._do_request("GET", "stock")
+        if parsed_json:
+            return [CurrentStockResponse(response) for response in parsed_json]
 
     def get_volatile_stock(self) -> CurrentVolatilStockResponse:
-        req_url = urljoin(self._base_url, "stock/volatile")
-        resp = requests.get(req_url, verify=self._verify_ssl, headers=self._headers)
-        parsed_json = resp.json()
-        return CurrentVolatilStockResponse(parsed_json)
+        parsed_json = self._do_request("GET", "stock/volatile")
+        if parsed_json:
+            return CurrentVolatilStockResponse(parsed_json)
```

Comment:
If the response is empty here, is this an error or expcted behaviour? If it is an error it should not be catched by using an if check.

Reply:
It cannot be an empty response.
Errors will be raise by raise_for_status().